java/arcs/core/allocator/Arc.kt

```diff
@@ -183,7 +183,11 @@ class Arc internal constructor(
 
     /** Stop the current [Arc]. */
     suspend fun stop() = allocator.stopArc(id).also {
-        onArcStateChangeFiltered(ArcState.Stopped) { closeFlow() }
+        onArcStateChangeFiltered(ArcState.Stopped) {
+            if (::closeFlow.isInitialized) {
+                closeFlow()
+            }
```

Comment:
I think it's generally frowned upon to use `isInitialized`.. maybe it's best to just use a nullable field at this point anyway.

Reply:
Removed, just added a no-op lambda as an initializer